sklearn/model_selection/tests/test_validation.py

```diff
@@ -1267,6 +1347,18 @@ def test_cross_val_predict_class_subset():
                                                         est, method)
         assert_array_almost_equal(expected_predictions, predictions)
 
+    # Special test for decision_function. This makes sure not to trigger
+    # any of the numerous edge cases in decision_function
+    X = np.arange(200).reshape(100, 2)
```

Comment:
I think you should keep decision_function in the loop above. Just find a dataset that works for decision_function. e.g. this one or one with many classes to avoid the edge cases that error.